lib/karafka/server.rb

```diff
@@ -0,0 +1,46 @@
+module Karafka
+  # Karafka consuming server class
+  class Server
+    class << self
+      # Method which runs app
+      def run
+        bind_on_sigint
+        bind_on_sigquit
+        start_supervised
+      end
+
+      private
+
+      # @return [Karafka::Process] process wrapper instance used to catch system signal calls
+      def process
+        Karafka::Process.instance
+      end
+
+      # What should happen when we decide to quit with sigint
+      def bind_on_sigint
+        process.on_sigint do
+          Karafka::App.stop!
+          exit
+        end
+      end
+
+      # What should happen when we decide to quit with sigquit
+      def bind_on_sigquit
+        process.on_sigquit do
```

Comment:
it is redundant on purpose. If we would add more siqnal handlers than each of them would be separated